pkg/minishift/docker/image/image_handler.go

```diff
@@ -0,0 +1,348 @@
+		return nil
+	}
+
+	session, err := handler.createSshSession()
+	if err != nil {
+		return err
+	}
+	defer session.Close()
+
+	r, err := session.StdoutPipe()
+	if err != nil {
+		return err
+	}
+
+	errorCh := make(chan error)
+	go func() {
+		file, err := os.Create(imageCacheFile + ".tmp")
+		if err != nil {
+			errorCh <- errors.New(fmt.Sprintf("Unable to create temporary image file: %v", err))
+		}
+
+		w := bufio.NewWriter(file)
+		io.Copy(w, r)
+
+		w.Flush()
+		file.Close()
+
+		err = os.Rename(file.Name(), imageCacheFile)
+		if err != nil {
+			errorCh <- errors.New(fmt.Sprintf("Unable to create image file: %v", err))
+		}
+
+		errorCh <- nil
+	}()
+
+	cmd := fmt.Sprintf("docker save %s", image)
+	err = session.Run(cmd)
+	if err != nil {
+		return errors.New(fmt.Sprintf("Error running command '%s': %v", cmd, err))
+	}
+
+	err = <-errorCh
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (handler *DockerImageHandler) pullImage(image string, w io.Writer) error {
+	defer timeTrack(time.Now(), w)
+	fmt.Fprint(w, fmt.Sprintf("Pulling image %s ", image))
+
+	session, err := handler.createSshSession()
+	if err != nil {
+		return err
+	}
+	defer session.Close()
+
+	cmd := fmt.Sprintf("docker pull %s", image)
+	err = session.Run(cmd)
+	if err != nil {
+		return errors.New(fmt.Sprintf("Error running command '%s': %v", cmd, err))
+	}
+
+	return nil
+}
+
+// availableImages returns a slice of available images in the current Docker instance
+func (handler *DockerImageHandler) availableImages() ([]string, error) {
+	session, err := handler.createSshSession()
+	if err != nil {
+		return nil, err
+	}
+	defer session.Close()
+
+	cmd := "docker images --format '{{.Repository}}:{{.Tag}}'"
+	var buffer bytes.Buffer
+	session.Stdout = &buffer
+	err = session.Run(cmd)
+	if err != nil {
+		return nil, errors.New(fmt.Sprintf("Error running command '%s': %v", cmd, err))
+	}
+
+	availableImages := strings.Split(buffer.String(), "\n")
+	return availableImages, nil
+}
+
+// imageNameToFileName maps an image name in the format repository/name:tag to a file system name.
+func (handler *DockerImageHandler) imageNameToFileName(image string) string {
+	imageNameParts := strings.FieldsFunc(image, func(r rune) bool {
+		switch r {
+		case '/', ':':
+			return true
+		}
+		return false
+	})
+
+	return strings.Join(imageNameParts, "@")
+}
+
+// isImageFile returns true if the denoted path represents a image file (based on file name).
+func (handler *DockerImageHandler) isImageFile(path string) bool {
+	fileName := filepath.Base(path)
+	match, err := regexp.Match("^.*@.*@.*$", []byte(fileName))
+	if err != nil {
+		return false
+	}
+	return match
+}
+
+// fileNameToImageName maps a filename on the host to an image name in the format repository/name:tag.
+func (handler *DockerImageHandler) fileNameToImageName(path string) string {
+	fileName := filepath.Base(path)
+	parts := strings.Split(fileName, "@")
+
+	return parts[0] + "/" + parts[1] + ":" + parts[2]
```

Comment:
Magic values... that have a meaning in other parts of the code. Please provide constant values to aid in possible refactor or readability when a change is needed.